Elf/__init__.py

```diff
@@ -907,12 +900,44 @@ def getDynSymTabInfo(self):
         in self.dynsymtabct.  Perhaps this is horrible and should be stricken
         from the code.
         '''
-        #FIXME: make this use both SECTION and PT_DYNAMICS versions...
+        # if DT_SYMTAB doesn't exist, we don't have a symbol table defined by DYNAMICS
         symtabva = self.dyns.get(DT_SYMTAB)
         if symtabva is None:
             return None, None, None
 
         symsz = self.dyns.get(DT_SYMENT)
+
+        # calculate/reuse symbol count
+        if not self.dynsymtabct:    # check if we've already got this value
+            # calculate the Dynamic Symbol Table Size
+            self.dynsymtabct = len(self.dynstrtab)  # cheat: there is a 1-to-1 relationship between symbols and strings in these tables
+            if not self.dynsymtabct:
+                # if we don't 
+                self._parseDynStrs()
+                self.dynsymtabct = len(self.dynstrtab)  # cheat: there is a 1-to-1 relationship between symbols and strings in these tables
+
+            # if "DT_SONAME" is within this string table, there are no symbols to match that or thereafter:
+            soname = self.dyns.get(DT_SONAME)
+            if soname is not None and soname != -1:
+                strsz = self.dyns.get(DT_STRSZ)
+                if soname < strsz:
+                    dynstrtabva = self.dyns.get(DT_STRTAB)
+                    strtabbytes = self.readAtRva(dynstrtabva, soname)
+                    dynsymstrs = strtabbytes.split(b'\0')
+                    self.dynsymtabct = len(dynsymstrs) - 1
+
+            # if any of the other Dyns which are addresses come after this, don't parse past them
+            for dyntype in dt_rebase:
```

Comment:
Where is dt_rebase declared? I can't find any mention of it anywhere else in this file.

Reply:
Nvm, I found dt_rebase in your vstruct change.  See my comment there.